Projects/Market/Sources/MarketPicker.swift

```diff
@@ -1,217 +1,140 @@
-import Apollo
-import Flow
-import Form
-import Foundation
-import Hero
-import Kingfisher
-import Presentation
-import UIKit
+import SwiftUI
 import hAnalytics
 import hCore
 import hCoreUI
-import hGraphQL
 
-public struct MarketPicker {
-    @Inject var client: ApolloClient
-    public init() {}
-}
-
-extension MarketPicker: Presentable {
-    public func materialize() -> (UIViewController, Signal<Void>) {
-        let viewController = UIViewController()
-
-        let appearance = UINavigationBarAppearance()
-        DefaultStyling.applyCommonNavigationBarStyling(appearance)
-        appearance.configureWithTransparentBackground()
-        viewController.navigationItem.standardAppearance = appearance
-        viewController.navigationItem.compactAppearance = appearance
+public struct MarketPickerView: View {
+    @ObservedObject var viewModel = MarketPickerViewModel()
+    @PresentableStore var store: MarketStore
 
-        let bag = DisposeBag()
+    @State var title: String = L10n.MarketLanguageScreen.title
+    @State var buttonText: String = L10n.MarketLanguageScreen.continueButtonText
 
-        ApplicationState.preserveState(.marketPicker)
-
-        let backgroundImageView = UIImageView()
+    @Environment(\.presentationMode) var presentationMode
```

Comment:
Not sure you should use presentaitonMode here, delegate this to the journey instead maybe?